osu.Game/Screens/Select/FooterButton.cs

```diff
@@ -142,5 +161,34 @@ protected override bool OnKeyDown(KeyDownEvent e)
 
             return base.OnKeyDown(e);
         }
+
+        protected class ScalingDrawSizePreservingFillContainer : DrawSizePreservingFillContainer
+        {
+            private readonly bool applyUIScale;
+            private Bindable<float> uiScale;
+
+            public override bool ReceivePositionalInputAt(Vector2 screenSpacePos) => true;
+
+            public ScalingDrawSizePreservingFillContainer(bool applyUIScale)
+            {
+                this.applyUIScale = applyUIScale;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(OsuConfigManager osuConfig)
+            {
+                if (applyUIScale)
+                {
+                    uiScale = osuConfig.GetBindable<float>(OsuSetting.UIScale);
+                    uiScale.BindValueChanged(scaleChanged, true);
+                }
+            }
+
+            private void scaleChanged(ValueChangedEvent<float> args)
+            {
+                this.ScaleTo(new Vector2(args.NewValue), 500, Easing.Out);
```

Comment:
probably needs a local comment explaining why this is done this way / required. since i'm not sure how this is working by reading your code.